UNL-CSE-AMBASSADORS/VITA
# 74 appt reschedule

Props to @CollinsSpencer for redoing this to fit the restyle and redoing the signup to use Angular.



This is technically done, but I'm not sure everything I did is "best practice" or "the better way to do things". @CollinsSpencer there's a few things I'm referring to in particular:


* Another thing... so the search functionality really only makes sense if it can search on _full_ first and last name... But if a volunteer doesn't have the 'view all client information' permission, then they could theoretically only search on first name and last initial, which isn't super intuitive and not very useful. What do we do about this? 
  * Do I just truncate the last name on the front-end for display and leave full last name for searching? The only problem is that this discloses the full last name client-side which is what we wanted to avoid in general, but idk.


**TESTING**
* Log in as a preparer, you should see a manage dropdown in the nav bar that only has appointments
* Log in as a site admin, you should see the manage dropdown in the nav bar that has appointments and the two others
* Navigate to /management/appointments
* Search for Barry, it should appear
* Click on the Barry appointment, it should go to a different page that has all the information and the ability to reschedule the appointment
* Select date/site/time, all that logic and stuff should work
* Hit reschedule, and the reschedule should be successful


**CHANGES TO PROD DB**
The following script needs to be run against the prod database when this is deployed:
```
ALTER TABLE Site
ADD COLUMN doesMultilingual BOOLEAN NOT NULL DEFAULT FALSE,
ADD COLUMN doesInternational BOOLEAN NOT NULL DEFAULT FALSE;

UPDATE TABLE Site
doesInternational = TRUE
WHERE siteId = 4;

UPDATE TABLE Site
doesMultilingual = TRUE
WHERE siteId = 3;
```
